tests/devices/qubit/test_sampling.py

```diff
@@ -40,6 +42,47 @@ def _init_state(n):
     return _init_state
 
 
+def _valid_flex_int(s):
+    """Returns True if s is a non-negative integer."""
+    return isinstance(s, int) and s >= 0
+
+
+def _valid_flex_tuple(s):
+    """Returns True if s is a tuple of the form (shots, copies)."""
+    return (
+        isinstance(s, tuple)
+        and len(s) == 2
+        and _valid_flex_int(s[0])
+        and isinstance(s[1], int)
+        and s[1] > 0
+    )
+
+
+class _FlexShots(Shots):  # pylint: disable=too-few-public-methods
```

Comment:
is the version defined in the source missing some stuff that was needed here?

Reply:
Fixed